test/extended/networking/network_segmentation_endpointslice_mirror.go

```diff
@@ -0,0 +1,321 @@
+			nadClient nadclient.K8sCniCncfIoV1Interface
+		)
+
+		BeforeEach(func() {
+			cs = f.ClientSet
+
+			var err error
+			nadClient, err = nadclient.NewForConfig(f.ClientConfig())
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		DescribeTable(
+			"mirrors EndpointSlices managed by the default controller for namespaces with user defined primary networks",
+			func(
+				netConfigParams networkAttachmentConfigParams,
+				isHostNetwork bool,
+			) {
+				netConfig := newNetworkAttachmentConfig(netConfigParams)
+
+				netConfig.namespace = f.Namespace.Name
+
+				By("creating the attachment configuration")
+				_, err := nadClient.NetworkAttachmentDefinitions(f.Namespace.Name).Create(
+					context.Background(),
+					generateNAD(netConfig),
+					metav1.CreateOptions{},
+				)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("deploying the backend pods")
+				replicas := 3
+				for i := 0; i < replicas; i++ {
+					runUDNPod(cs, f.Namespace.Name,
+						*podConfig(fmt.Sprintf("backend-%d", i), func(cfg *podConfiguration) { cfg.namespace = f.Namespace.Name }),
+						func(pod *corev1.Pod) {
+							pod.Spec.HostNetwork = isHostNetwork
+							if pod.Labels == nil {
+								pod.Labels = map[string]string{}
+							}
+							pod.Labels["app"] = "test"
+						})
+				}
+
+				By("creating the service")
+				svc := e2eservice.CreateServiceSpec("test-service", "", false, map[string]string{"app": "test"})
+				familyPolicy := corev1.IPFamilyPolicyPreferDualStack
+				svc.Spec.IPFamilyPolicy = &familyPolicy
+				_, err = cs.CoreV1().Services(f.Namespace.Name).Create(context.Background(), svc, metav1.CreateOptions{})
+				framework.ExpectNoError(err, "Failed creating service %v", err)
+
+				nodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{LabelSelector: "node-role.kubernetes.io/worker"})
+				framework.ExpectNoError(err, "Failed listing worker nodes %v", err)
+				Expect(len(nodes.Items)).To(BeNumerically(">", 0))
+
+				v4HostSubnet, v6HostSubnet, err := parseHostSubnet(&nodes.Items[0])
+				framework.ExpectNoError(err, "Failed parsing host subnet %v", err)
+				// In cloud use the EIP annotation, this is especially useful in GCP where the hostSubnet has the `/32` mask
+				if _, ok := nodes.Items[0].Annotations[nodeEgressIPConfigAnnotationKey]; ok {
+					nodeConfig, err := getNodeEgressIPConfiguration(&nodes.Items[0])
+					framework.ExpectNoError(err, "Failed getting node configuration %v", err)
+					Expect(nodeConfig).To(HaveLen(1))
+					v4HostSubnet, v6HostSubnet = nodeConfig[0].IFAddr.IPv4, nodeConfig[0].IFAddr.IPv6
+				}
+
+				isDualStack := v4HostSubnet != "" && v6HostSubnet != ""
+
+				By("asserting the mirrored EndpointSlice exists and contains PODs primary IPs")
+				Eventually(func() error {
+					return validateMirroredEndpointSlices(cs, f.Namespace.Name, svc.Name, userDefinedNetworkIPv4Subnet, userDefinedNetworkIPv6Subnet, v4HostSubnet, v6HostSubnet, replicas, isDualStack, isHostNetwork)
+
```

Comment:
nit: this extra line is confusing. either drop it, or put all `validateMirroredEndpointSlices` args in a separate line